File: src/v2/core/src/com/mygdx/game/view/DefaultStrategy.cs
namespace Client.View
{
    // should be invoked at the end of strategies
    public class DefaultStrategy : ITileStrategy
    {
        private readonly TextureHolder texture;
        private object p;

        public DefaultStrategy(TextureHolder texture)
        {
            this.texture = texture;
        }

        public bool CanExecute(LocationType[] neighbors)
        {
            return true;
        }

        public TextureHolder Execute(LocationType[] neighbors)
        {
            return texture;
        }
    }
}

File: src/core/src/com/mygdx/game/view/DefaultStrategy.java
package com.mygdx.game.view;

// should be invoked at the end of strategies
public class DefaultStrategy implements ITileStrategy {
    private final TextureHolder texture;

    public DefaultStrategy(TextureHolder texture) {
        this.texture = texture;
    }

    public boolean CanExecute(LocationType[] neighbors) {
        return true;
    }

    public TextureHolder Execute(LocationType[] neighbors) {
        return texture;
    }
}
